projects/igniteui-angular/src/lib/combo/combo.common.ts

```diff
@@ -27,6 +53,1244 @@ export interface IgxComboBase {
+     * // get
+     * let myComboWidth = this.combo.width;
+     * ```
+     *
+     * ```html
+     * <!--set-->
+     * <igx-combo [width]='250px'></igx-combo>
+     * ```
+     */
+    @HostBinding('style.width')
+    @Input()
+    public width: string;
+
+    /**
+     * Controls whether custom values can be added to the collection
+     *
+     * ```typescript
+     * // get
+     * let comboAllowsCustomValues = this.combo.allowCustomValues;
+     * ```
+     *
+     * ```html
+     * <!--set-->
+     * <igx-combo [allowCustomValues]='true'></igx-combo>
+     * ```
+     */
+    @Input()
+    public allowCustomValues = false;
+
+    /**
+     * Configures the drop down list height
+     *
+     * ```typescript
+     * // get
+     * let myComboItemsMaxHeight = this.combo.itemsMaxHeight;
+     * ```
+     *
+     * ```html
+     * <!--set-->
+     * <igx-combo [itemsMaxHeight]='320'></igx-combo>
+     * ```
+     */
+    @Input()
+    public get itemsMaxHeight(): number {
+        if (this._itemsMaxHeight === null || this._itemsMaxHeight === undefined) {
+            return this.itemHeight * itemsInContainer;
+        }
+        return this._itemsMaxHeight;
+    }
+
+    public set itemsMaxHeight(val: number) {
+        this._itemsMaxHeight = val;
+    }
+
+    /**
+     * Configures the drop down list item height
+     *
+     * ```typescript
+     * // get
+     * let myComboItemHeight = this.combo.itemHeight;
+     * ```
+     *
+     * ```html
+     * <!--set-->
+     * <igx-combo [itemHeight]='32'></igx-combo>
+     * ```
+     */
+    @Input()
+    public get itemHeight(): number {
+        if (this._itemHeight === null || this._itemHeight === undefined) {
+            return ItemHeights[this.displayDensity];
+        }
+        return this._itemHeight;
+    }
+
+    public set itemHeight(val: number) {
+        this._itemHeight = val;
+    }
+
+    /**
+     * Configures the drop down list width
+     *
+     * ```typescript
+     * // get
+     * let myComboItemsWidth = this.combo.itemsWidth;
+     * ```
+     *
+     * ```html
+     * <!--set-->
+     * <igx-combo [itemsWidth] = '"180px"'></igx-combo>
+     * ```
+     */
+    @Input()
+    public itemsWidth: string;
+
+    /**
+     * Defines the placeholder value for the combo value field
+     *
+     * ```typescript
+     * // get
+     * let myComboPlaceholder = this.combo.placeholder;
+     * ```
+     *
+     * ```html
+     * <!--set-->
+     * <igx-combo [placeholder]='newPlaceHolder'></igx-combo>
+     * ```
+     */
+    @Input()
+    public placeholder;
+
+    /**
+     * Defines the placeholder value for the combo dropdown search field
+     *
+     * ```typescript
+     * // get
+     * let myComboSearchPlaceholder = this.combo.searchPlaceholder;
+     * ```
+     *
+     * ```html
+     * <!--set-->
+     * <igx-combo [searchPlaceholder]='newPlaceHolder'></igx-combo>
+     * ```
+     */
+    @Input()
+    public searchPlaceholder = 'Enter a Search Term';
```

Comment:
this makes no sense to go in common. Search placeholder is used in the drop down input of the `igx-combo`. Simple combo has no such input. Now setting `searchPlaceholder` for simple combo works as `placeholder`. However setting both `searchPlaceholder` and `placeholder` will end up showing the `placeholder`.